metricbeat/helper/server/http/http.go

```diff
@@ -44,6 +44,12 @@ type HttpEvent struct {
 	meta  server.Meta
 }
 
+// MetricSetFactory accepts a BaseMetricSet and returns a MetricSet. If there
+// was an error creating the MetricSet then an error will be returned. The
+// returned MetricSet must also implement either EventFetcher or EventsFetcher
+// (but not both).
+type HandlerFactory func(h *HttpServer) func(writer http.ResponseWriter, req *http.Request)
```

Comment:
Do you really need a factory? The metricset is already initializing the server, so you could pass a private `metricset.handleFunc` which implements `func(writer http.ResponseWriter, req *http.Request)`